### ISSUE TITLE
Input of Densenet

### ISSUE THREAD

**Issue by mangdian** [kenshohara/3D-ResNets-PyTorch#27]
Thank you for your wonderful work.
Just read the paper, it is noted that each clip contains 16 frames. I read two other papers in which the author claims that 32 frame input would be better, have you tried 32 frames input? If you trained such models, can you please release the pretrained models?

**Comment by kenshohara** [comment]
You are right.
Larger input size improves action recognition accuracies.
I tried only ResNeXt-101 with 64 frame input.
Kinetics pretrained and UCF-101 fine-tuned ResNeXt-101 (64f) is released.
```resnext-101-kinetics-ucf101_split1.pth``` is the one.

**Comment by kenshohara** [comment]
Input sizes of all our pretrained models are 16x112x112.
I think larger input size achieves better accuracies
but it requires large GPU memories.

**Comment by mangdian** [comment]
That's True. Besides, I read the code of your 3d DenseNet model and found that you used 2d bn instead of 3d bn. Why?

**Comment by kenshohara** [comment]
I also found that and will fix it.
If you use input frame size larger than 16, an error may occur.

**Comment by mangdian** [comment]
@kenshohara So in your pretrained model, the bn layer is 2d instead of 3d?  
I have not run the code. However, in the densenet.py file, you write these code:
`last_duration = int(math.ceil(self.sample_duration / 16))
        last_size = int(math.floor(self.sample_size / 32))
        out = F.avg_pool3d(
            out, kernel_size=(last_duration, last_size, last_size)).view(
                features.size(0), -1)
        out = self.classifier(out)`
which is used to pool different input into the same size. larger size than 16 may be OK?

**Comment by mangdian** [comment]
@kenshohara  Just want to make sure that whether you use 2d bn in training or it is just a small mistake in the code

**Comment by kenshohara** [comment]
I used 2d bn when I trained the model.
I think it is not a problem because the difference between BatchNorm2d and BatchNorm3d is _check_input_dim.
http://pytorch.org/docs/master/_modules/torch/nn/modules/batchnorm.html#BatchNorm3d

**Comment by mangdian** [comment]
But the parameters may not work when I replace 2d bn with 3d bn

**Comment by kenshohara** [comment]
No, the parameters of bn 3d is the same as those of bn 2d.
Please see source codes.
